controllers/che/checluster_controller.go

```diff
@@ -260,6 +260,29 @@ func (r *CheClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		CheCluster: instance,
 	}
 
+	if isPreviousVersionDeployed(instance) {
+		// Current operator is newer than deployed Che
+		backupCR, err := getBackupCR(deployContext)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Create a backup before updating current installation
+				if err := requestNewBackup(deployContext); err != nil {
```

Comment:
If we do backup on every update, will be enough storage for it?
How can we clean up previous backups?

Reply:
For now we do not backup users' projects, so the backup itself is tiny. But when we include user projects in the backup we should think about clean up. I think backups cleanup is a separate topic, thought.